### PR TITLE
send cabinet office ggroup data to splunk

### PR DESCRIPTION
**Pull request by alice-carr**
this pr:
- adds a deploy co job to the concourse pipeline
- adds a new terraform deployment for cabinet office
- passes in a suffix to differentiate cabinet office terraform

also as a fly by:
- change underscores in pipeline to hyphens

relates #23

### CODE REVIEW

**Review by denizgenc — APPROVED** [review]
Looks good, and thanks for changing job names from snake case to kebab case

File: terraform/modules/ggroup-to-splunk/variables.tf [hunk 1]
```diff
@@ -24,3 +24,14 @@ variable "environment" {
 variable "lambda_role_name" {
   type = string
 }
+
```
Comment (akinnane) [inline terraform/modules/ggroup-to-splunk/variables.tf:27]:
```suggestion
```

File: terraform/modules/ggroup-to-splunk/variables.tf [hunk 2]
```diff
@@ -24,3 +24,14 @@ variable "environment" {
...
+variable "credentials_prefix" {
+  type    = string
+  default = ""
+}
+
+variable "suffix" {
+  type    = string
+  default = ""
+}
```
Comment (akinnane) [inline terraform/modules/ggroup-to-splunk/variables.tf]:
```suggestion
}

```

File: concourse/pipeline.yml [hunk 3]
```diff
@@ -204,10 +206,54 @@ jobs:
...
+        on_success:
+          <<: *health_status_notify
+          params:
+            message: "Ggroup data to splunk prod co deployed successfully."
             health: healthy
         on_failure:
           <<: *health_status_notify
           params:
-            message: "Failed to deploy Ggroup data to splunk validation failed dev."
+            message: "Failed to deploy Ggroup data to splunk failed to deploy to prod co."
```
Comment (akinnane) [inline concourse/pipeline.yml]: It this a sentence?

**Review by danjoneslf — APPROVED** [review]
Looks good. What does the dev deploy do creds-wise? Should we plug that into the cabinet-office.co.uk domain if we haven't already?

File: concourse/pipeline.yml (hunk 3 above)
Comment (alice-carr) [reply concourse/pipeline.yml]: it doesn't look like one in hindsight. prod co here stands for prod cabinte office but I see how that's not clear

Comment (alice-carr) [reply concourse/pipeline.yml]: wait I just reread that whole sentence and it's worse than I thought

**Comment by alice-carr** [comment]
> Looks good. What does the dev deploy do creds-wise? Should we plug that into the cabinet-office.co.uk domain if we haven't already?

@danjoneslf I believe dev and prod use the same creds (for some reason i don't have access to demo to check) but it feeds to a logstream that's linked to splunk's test_data index. What do you mean by Should we plug that into the cabinet-office.co.uk domain if we haven't already?

**Pull request merged**